data_override/data_override.F90

```diff
@@ -569,120 +134,93 @@ subroutine data_override_unset_domains(unset_Atm, unset_Ocean, &
-  if(domain .EQ. NULL_DOMAIN2D) call mpp_error(FATAL,'data_override: failure in get_domain')
-  if(present(comp_domain)) &
-     call mpp_get_compute_domain(domain,comp_domain(1),comp_domain(2),comp_domain(3),comp_domain(4))
-end subroutine get_domain
-
-!> @brief Given a gridname, this routine returns the working domain associated with this gridname
-subroutine get_domainUG(gridname, UGdomain, comp_domain)
-  character(len=3), intent(in) :: gridname
-  type(domainUG), intent(inout) :: UGdomain
-  integer, intent(out), optional :: comp_domain(4) !< istart,iend,jstart,jend for compute domain
-  type(domain2D), pointer :: SGdomain => NULL()
+!> @brief Routine to perform data override for scalar fields
+subroutine data_override_0d(gridname,fieldname_code,data,time,override,data_index)
+  character(len=3), intent(in) :: gridname !< model grid ID (ocn,ice,atm,lnd)
+  character(len=*), intent(in) :: fieldname_code !< field name as used in the model (may be
+                                                 !! different from the name in NetCDF data file)
+  logical, intent(out), optional :: override !< true if the field has been overriden succesfully
+  type(time_type), intent(in) :: time !< (target) model time
+  class(*),             intent(out) :: data !< output data array returned by this call
+  integer, intent(in), optional :: data_index
 
-  UGdomain = NULL_DOMAINUG
-  select case (gridname)
-     case('LND')
-        UGdomain = lnd_domainUG
-     case default
-        call mpp_error(FATAL,'error in data_override get_domain')
+  select type(data)
+    type is (real(r4_kind))
+      call data_override_0d_r4(gridname,fieldname_code,data,time,override,data_index)
+    type is (real(r8_kind))
+      call data_override_0d_r8(gridname,fieldname_code,data,time,override,data_index)
+    class default
+      call mpp_error(FATAL, "data_override_0d: Unsupported data type")
   end select
-!  if(UGdomain .EQ. NULL_DOMAINUG) call mpp_error(FATAL,'data_override: failure in get_domain')
-  if(present(comp_domain)) &
-     call mpp_get_UG_SG_domain(UGdomain,SGdomain)
-     call mpp_get_compute_domain(SGdomain,comp_domain(1),comp_domain(2),comp_domain(3),comp_domain(4))
-end subroutine get_domainUG
-!===============================================================================================
+end subroutine data_override_0d
 
 !> @brief This routine performs data override for 2D fields; for usage, see data_override_3d.
 subroutine data_override_2d(gridname,fieldname,data_2D,time,override, is_in, ie_in, js_in, je_in)
   character(len=3), intent(in) :: gridname !< model grid ID
   character(len=*), intent(in) :: fieldname !< field to override
   logical, intent(out), optional :: override !< true if the field has been overriden succesfully
   type(time_type), intent(in) :: time !<  model time
-  real, dimension(:,:), intent(inout) :: data_2D !< data returned by this call
+  class(*), dimension(:,:), intent(inout) :: data_2D !< data returned by this call
```

Comment:
This is still not fitting the style of the project using macros.

Reply:
This subroutine was removed in b89c8db621164eef22e0d9220a2046a4a1ceaa98.